core/src/main/scala/kafka/admin/AclCommand.scala

```diff
@@ -109,7 +141,39 @@ object AclCommand extends Logging {
     }
   }
 
+  private def addAclWithAdminAPI(opts: AclCommandOptions): Unit = {
+    withAdminClient(opts) { adminClient =>
```

Comment:
What if we created an object that encapsulated either an AdminClient object or an Authorizer object?  The current mechanism of duplicating all the code paths in withAdminAPI and withAuthorizer variants leads to some duplicated code.

Reply:
@cmccabe Thanks for the review. Yes, we encapsulate the logic in different classes.
Updated the PR. Pls take look.

cc @hachikuji